extensions/src/org/pentaho/platform/plugin/services/security/userrole/ldap/DefaultLdapUserRoleListService.java

```diff
@@ -155,6 +165,13 @@ public void afterPropertiesSet() throws Exception {
     for ( GrantedAuthority role : results ) {
       roles.add( role.getAuthority() );
     }
+    // Now add extra role if it does not exist in the list
+    for ( String extraRole : extraRoles ) {
```

Comment:
Whoa ... just because there are extra roles doesn't mean that this user has all of those roles, does it ? Is this correct ?